components/dependency_injection.rst

```diff
@@ -290,17 +290,21 @@ config files:
 
     .. code-block:: php
 
-        use Symfony\Component\DependencyInjection\Reference;
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+        return function(ContainerConfigurator $configurator) {
+            $configurator->parameters()
+                ->set('mailer.transport', 'sendmail');
+
+            $container = $configurator->services();
```

Comment:
I am not whether we should go for this or:
```php
return function (ContainerConfigurator $container) {
    $container->parameters()
        ->set('mailer.transport', 'sendmail')
        ->set('xxx', 'XXX')
    ;

    $container->services()
        ->set(...)->args(...)
        ->set(...)->call()->public()

        ->instancof()->tags()
        // ...
```
WDYT?
There are so many paths. Playing with this is a real joy, I guess I won't be able to use any other format now :).
I would like we all agree on a best practice for this to roll it out.
ping @symfony/team-symfony-docs @nicolas-grekas

Reply:
K, so I actually have good use case for why you should keep the master `$configurator` separate from the services configurator named: `$container` .

There are times when extending an abstract service that you cannot be within the context of a service configurator with defaults set. So you need to keep a reference to the original ContainerConfigurator instance to do so.

Reply:
@HeahDude not sure it matters to much, I think I personally prefer a separate statement per definition just so that it makes the definitions more distinct in the code, but people can also style their stuff however they want.